src/base/bittorrent/session.cpp

```diff
@@ -4573,6 +4573,7 @@ void Session::startUpTorrents()
     int resumedTorrentsCount = 0;
     QVector<TorrentID> queue;
     QSet<QString> recoveredCategories;
+    QSet<QString> recoveredTags;
```

Comment:
Are there real usage of it? It looks redundant. It seems you can just use `Session::hasTag()` instead of this variable.